src/linux/net/wifi/core/AccessPointControllerLinux.cxx

```diff
@@ -366,6 +448,18 @@ AccessPointControllerLinux::SetAkmSuites(std::vector<Ieee80211AkmSuite> akmSuite
         return status;
     }
 
+    // Validate that the AKM suites are successfully set by hostapd.
+    auto hostapdBssConfiguration = m_hostapd.GetConfiguration();
+    auto actualKeyManagements = hostapdBssConfiguration.WpaKeyMgmt;
+    if (std::equal(std::begin(wpaKeyManagements), std::end(wpaKeyManagements), std::begin(actualKeyManagements))) {
+        AUDIT_LOGI << std::format("AKM suites of AP {} set to {}", status.AccessPointId, akms);
+    } else {
+        AUDIT_LOGD << std::format("AKM suites of AP {} were not set. Current key_mgmt values:", status.AccessPointId);
+        for (const auto& actualKeyManagement : actualKeyManagements) {
+            AUDIT_LOGD << std::format("key_mgmt: {}", std::to_underlying(actualKeyManagement));
```

Comment:
This will have one entry per actual key management value set. Should we instead create an aggregated string that has all of them so we can add a single audit log entry?